### PR TITLE
Fix #1657 - Make gateway jwt cache configurable

### LINKED ISSUE #1657: Make jwt token cache configurable
### Description:
Microgateway uses node local cache in order to validate the jwts. The caching configs should be configurable.

Following is the config section
```toml
[enforcer.cache]
  enabled = true
  maximumSize = 10000
  expiryTime = 15    
```

### Affected Product Version:
4.0.0-m6

### Environment details (with versions):
- OS:
- Client:
- Env (Docker/K8s):

---

### PR DESCRIPTION
**Pull request by Rajith90**
### Purpose
Following config section is added.

```toml
[enforcer.cache]
  enabled = true
  maximumSize = 10000
  expiryTime = 15   
```

### Issues

Fixes #1657 

### Automation tests
 - Unit tests added: Yes/No
 - Integration tests added: Yes/No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

File: adapter/config/mgwtypes.go [hunk 1]
```diff
@@ -191,6 +192,12 @@ type jwtGenerator struct {
 	PrivateKeyPath        string `toml:"privateKeyPath"`
 }
 
+type cache struct {
+	Enabled               bool   `toml:"enabled"`
+	MaximumSize           int32  `toml:"maximumSize"`
+	ExpiryTime			  int32  `toml:"expiryTime"`
```
Comment (Rajith90) [inline adapter/config/mgwtypes.go:198]: This looks fine in the Intellij Idea

File: adapter/config/mgwtypes.go [hunk 2]
```diff
@@ -125,6 +125,7 @@ type Config struct {
 		ApimCredentials apimCredentials
 		AuthService     authService
 		JwtGenerator    jwtGenerator
+		Cache		    cache
```
Comment (Rajith90) [inline adapter/config/mgwtypes.go:128]: This looks fine in the Intellij Idea

File: enforcer/src/main/java/org/wso2/micro/gateway/enforcer/security/jwt/JWTAuthenticator.java [hunk 3]
```diff
@@ -493,4 +496,14 @@ private SignedJWTInfo getSignedJwt(String accessToken) throws ParseException {
         }
         return signedJWTInfo;
     }
+
+    private String getJWTTokenIdentifier(SignedJWTInfo signedJWTInfo) {
+
+        JWTClaimsSet jwtClaimsSet = signedJWTInfo.getJwtClaimsSet();
+        String jwtid = jwtClaimsSet.getJWTID();
+        if (org.apache.commons.lang.StringUtils.isNotEmpty(jwtid)) {
```
Comment (VirajSalaka) [inline enforcer/src/main/java/org/wso2/micro/gateway/enforcer/security/jwt/JWTAuthenticator.java:504]: How about using commons3 ?

File: adapter/config/mgwtypes.go (hunk 2 above)
Comment (praminda) [reply adapter/config/mgwtypes.go:128]: you have both tabs and spaces in here. thats why this has happened shall we stick to one? I think go recommends tab for indentation not spaces

File: adapter/config/mgwtypes.go (hunk 1 above)
Comment (praminda) [reply adapter/config/mgwtypes.go:198]: same here

File: api/wso2/discovery/config/enforcer/cache.proto [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+// [#protodoc-title: Cache]
+
+// JWT Generator model
+message Cache {
+    bool enable = 1;
+
+    int32 maximumSize = 2;
+
+    sint32 expiryTime = 3;
+}
```
Comment (praminda) [inline api/wso2/discovery/config/enforcer/cache.proto:19]:
```suggestion
}

```

File: adapter/config/mgwtypes.go (hunk 2 above)
Comment (SuKSW) [reply adapter/config/mgwtypes.go:128]: Opening the file in vs code, changing something, and pressing "CTRL+S" automatically fixes this.

**Review by praminda — APPROVED** [review]

**Pull request merged**